File: python/bob/bob.py
class Bob:
    "A lackadaisical teenager"

    def hey(self, says_the_other):
        "Bob's response to anything"
        if not says_the_other or says_the_other.isspace():
            return 'Fine. Be that way!'
        elif says_the_other.isupper():
            return 'Woah, chill out!'
        elif says_the_other[-1] == '?':
            return 'Sure.'
        return 'Whatever.'

File: go/bob/bob.go
package bob

import (
	"strings"
	"unicode"
)

func Hey(text string) string {
	switch {
	case isSilence(text):
		return "Fine. Be that way!"
	case isShouting(text):
		return "Woah, chill out!"
	case isQuestion(text):
		return "Sure."
	default:
		return "Whatever."
	}
}

func isSilence(text string) bool {
	return strings.IndexFunc(text, func(r rune) bool {
		return !unicode.IsSpace(r)
	}) == -1
}

func isShouting(text string) bool {
	return strings.IndexFunc(text, unicode.IsUpper) != -1 &&
		strings.IndexFunc(text, unicode.IsLower) == -1
}

func isQuestion(text string) bool {
	return strings.HasSuffix(text, "?")
}
